discovery/marathon/marathon.go

```diff
@@ -427,7 +428,7 @@ func targetsForApp(app *App) []model.LabelSet {
 		ports, labels = extractPortMapping(app.Container.Docker.PortMappings, app.isContainerNet())
 		prefix = portMappingLabelPrefix
 
-	} else if len(app.PortDefinitions) != 0 {
+	} else if app.RequirePorts && len(app.PortDefinitions) != 0 {
```

Comment:
This will effectively disable label gathering when `requirePorts` is false, which is the default. This should have broken at least a test or two, please take a look at this, I will dive in as well. This might be missing from coverage.

In any case, I would wrap `ports[i] = app.PortDefinitions[i].Port` below in an `if app.RequirePorts` instead of this. This will make sure `ports` is still initialized to an array of zeroes of the correct length in both cases, making your change on line 462 unnecessary. (a zero will cause a port lookup in the task)
Make sure to add a comment, eg.
```
// When requirePorts is false, this port becomes the 'servicePort', not the listen port.
// In this case, the port needs to be taken from the task instead of the app.
```